ggmap/analyses.py

```diff
@@ -4082,6 +4082,144 @@ def post_cache(cache_results):
                      array=1,
                      **executor_args)
 
+def ancom(counts: pd.DataFrame, rank, taxonomy: pd.Series, grouping: pd.Series, min_mean_abundance_per_feature: float=0.005,
+          ppn=1, pmem='4GB', **executor_args):
+    """Execute ANCOM analysis through Qiime2.
+
+    Paramaters
+    ----------
+    counts: pd.DataFrame
+        The feature table.
+    rank: str or (str, str)
+        A taxonomic rank to which features shall be collapsed.
+        Use 'raw' to avoid collapsing.
+        Use a tuple e.g. ('Family', 'Genus') to collapse on merged taxonomy ranks.
+    taxonomy: pd.Series
+        Taxonomic lineages for each feature in counts. Can be None if rank is 'raw'.
+    min_mean_abundance_per_feature: float
+        only report features, that have on average a relative abundance of min_mean_abundance_per_feature
+    palette: dict, optional
+        You can pass a color dictionary to the underlying sns.barplot function
+    executor_args:
+        dry, use_grid, nocache, wait, walltime, ppn, pmem, timing, verbose, dirty
+
+    Returns
+    -------
+    """
+
+    COL = 'COLANCOM'
+    # general sanity checks
+    (counts, grouping) = sync_counts_metadata(counts, grouping.dropna())
+    if grouping.value_counts().shape[0] < 2:
+        raise ValueError("Your grouping information has only ONE group. It must be exactly two!")
+    if grouping.value_counts().shape[0] > 2:
+        raise ValueError("Your grouping information more than two groups. It must be exactly two!")
```

Comment:
Typo: missing word "has"
```suggestion
        raise ValueError("Your grouping information has more than two groups. It must be exactly two!")
```

Reply:
fixed